sdk/containerregistry/azure-containerregistry/azure/containerregistry/_download_stream.py

```diff
@@ -0,0 +1,76 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import hashlib
+from typing import Iterator, ContextManager, TypeVar, cast, Tuple, Dict, Any
+from typing_extensions import Protocol
+
+T = TypeVar('T', bound='DownloadBlobStream')
+
+
+class GetNext(Protocol):
+    def __call__(self, *args: Any, range_header: str) -> Iterator[bytes]:
+        pass
+
+
+class DownloadBlobStream(
+    Iterator[bytes],
+    ContextManager[T],
+):
+    """Protocol for methods to provide streamed responses."""
+
+    def __init__(
+        self,
+        *,
+        response: Iterator[bytes],
+        get_next: GetNext,
+        blob_size: int,
+        downloaded: int,
+        digest: str,
+        chunk_size: int
+    ) -> None:
+        self._response = response
+        self._next = get_next
+        self._blob_size = blob_size
+        self._downloaded = downloaded
+        self._digest = digest
+        self._chunk_size = chunk_size
+        self._hasher = hashlib.sha256()
+
+    def __enter__(self: T) -> T:
+        return self
+
+    def __exit__(self, *args) -> None:
+        return None
```

Comment:
This must call close.

Reply:
Where did the close method go?
Now that the code regeneration is merged, the close method will need to be something like:
```python
def close(self):
    self._response.http_response.close()
```

Reply:
Doesn't look like this has been fixed yet?